client/network/src/service.rs

```diff
@@ -1141,6 +1191,10 @@ where
 			.to_worker
 			.unbounded_send(ServiceToWorkerMsg::NewBestBlockImported(hash, number));
 	}
+
+	fn on_block_finalized(&self, hash: B::Hash, header: B::Header) {
+		let _ = self.to_worker.unbounded_send(ServiceToWorkerMsg::BlockFinalized(hash, header));
+	}
```

Comment:
This should only be needed for `ChainSync`. There we also have some service struct. We could directly use this to forward the finalized block.